lms/djangoapps/courseware/access_utils.py

```diff
@@ -103,3 +112,57 @@ def check_course_open_for_learner(user, course):
     if COURSE_PRE_START_ACCESS_FLAG.is_enabled():
         return ACCESS_GRANTED
     return check_start_date(user, course.days_early_for_beta, course.start, course.id)
+
+
+def check_enrollment(user, course):
```

Comment:
So the spirit of these new check functions is similar to the ones above - they're each slices of a larger access question, effectively.  They can be read as "check enrollment-related conditions" and "check public access-related conditions"